src/test/scala/it/unibo/authsim/library/dsl/cryptography/encrypter/symmetric/SymmetricEncryptionTest.scala

```diff
@@ -22,13 +23,23 @@ class SymmetricEncryptionTest extends AnyWordSpec with Matchers with BeforeAndAf
       for(password<- passwordList; secret<- secretList)
         des.decrypt( des.encrypt(password, secret), secret).equals(password) shouldBe true
     }
+
+    "allow to re-define the salt value" in{
+      aes.algorithm.salt_(salt)
```

Comment:
This fails as `EncryptionAlgorithm` (from `Cipher.algorithm`) does not contain `salt_` method.

This can be resolved by an `asInstanceOf[AES]`, but I'd (and I'll) try to find a way to avoid using this workaround.

Reply:
I'm thinking about changing `Cipher` into `Cipher[T <: EncryptionAlgorithm]` and `def algorithm: EncryptionAlgorithm` into `def algorithm: T`, so every implementer of `Cipher` specifies the type returned by `algorithm` (the same thing has been done in the `Alphabet` trait, but here we don't even have a circular dependency, so it should be simpler).